services/AdminDashboardService.cfc

```diff
@@ -43,26 +51,43 @@ component {
 	}
 
 	public boolean function userCanShareDashboard( required string dashboardId, string adminUserId=$getAdminLoggedInUserId() ) {
+		if ( hasFullAccess( arguments.adminUserId ) ) {
+			return true;
+		}
+
 		return $getPresideObject( "admin_dashboard" ).dataExists(
 			filter = { id=arguments.dashboardId, owner=arguments.adminUserId }
 		);
 	}
 
 	public boolean function userCanDeleteDashboard( required string dashboardId, string adminUserId=$getAdminLoggedInUserId() ) {
+		if ( hasFullAccess( arguments.adminUserId ) ) {
+			return true;
+		}
+
 		return $getPresideObject( "admin_dashboard" ).dataExists(
 			filter = { id=arguments.dashboardId, owner=arguments.adminUserId }
 		);
 	}
 
+	public boolean function hasFullAccess( required string adminUserId ) {
+		var adminUserGroups = _getAdminUserGroups( arguments.adminUserId );
```

Comment:
Could we maybe use the PermissionService$userHasAssignedRoles() method? e.g.

`return permissionService.userHasAssignedRoles( userId=arguments.adminUserId, roles=[ "sysadmin" ] );`

Reply:
Yes agree we should use the PermissionService$userHasAssignedRoles()